lib/rot.rb

```diff
@@ -0,0 +1,38 @@
+class Rot
+  ALPHABET = (("a".."z").to_a + ("A".."Z").to_a).freeze
+  MAX_ROT_KEY = 26
+
+  def initialize(rot_key)
+    @rot_key = rot_key
+    if @rot_key > MAX_ROT_KEY
+      raise ArgumentError, "Given rot key must be less than 26."
+    end
+  end
+
+  private
+  
+  def encryptable_letter?(letter)
+    ALPHABET.include?(letter)
+  end
+
+  def encrypt_letter(letter)
+    if encryptable_letter?(letter) 
+      @rot_key.times do 
+      letter = letter.next
+      end
+    end
+    letter
+  end
+
+  public
+
+  def encrypt(message)
+    message = message.split("")
+    result = ""
+    for letter in message
+      encrypted_letter = encrypt_letter(letter)
+      result << encrypted_letter[-1]
```

Comment:
@djordje997 why do you have to use `[-1]`? I would expect `encrypt_letter` to return, well, an encrypted letter. I wouldn't expect to have to modify it further.

I would expect to be able to do away with the local variable and simply have a one-line like this:
```suggestion
      result << encrypt_letter(letter)
```